lib/elixir/test/elixir/typespec_test.exs

```diff
@@ -120,14 +120,46 @@ defmodule TypespecTest do
       end
     end
 
-    test "undefined spec for function" do
+    test "type variable only used once (singleton type variable)" do
+      assert_raise CompileError, ~r"type variable x is only used once", fn ->
```

Comment:
I think the error message is confusing because the type variable is never used. The declaration is not a use. I think we should try to match the warning we have for normal unused variables.

Reply:
Maybe there's a reason why they make a distinction between never used and only used once but I don't see it 🤔. I would go with `type variable x is unused`.